src/test/java/seedu/address/storage/JsonAdaptedNoteTest.java

```diff
@@ -3,23 +3,37 @@
 import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
+import java.util.HashSet;
+import java.util.Set;
+
 import org.junit.jupiter.api.Test;
 
+import seedu.address.commons.exceptions.IllegalValueException;
 import seedu.address.model.country.Country;
 import seedu.address.model.note.CountryNote;
 import seedu.address.model.note.Note;
+import seedu.address.model.tag.Tag;
 
 public class JsonAdaptedNoteTest {
-    @Test
-    public void toModelType_validClientNote_returnsClientNote() {
-        JsonAdaptedNote jsonAdaptedNote = new JsonAdaptedNote(new Note("some note"));
-        assertTrue(jsonAdaptedNote.toModelType().isClientNote());
+
+    @Test public void toModelType_validTaggedClientNote_returnsTaggedClientNote() throws IllegalValueException {
+        Note taggedClientNote = new Note("some note");
+        Set<Tag> associatedTags = new HashSet<>();
+        associatedTags.add(new Tag("someTag"));
+        taggedClientNote.setTags(associatedTags);
+        JsonAdaptedNote jsonAdaptedNote = new JsonAdaptedNote(taggedClientNote);
+        Note deserialisedNote = jsonAdaptedNote.toModelType();
+        assertTrue(deserialisedNote.isClientNote());
+        assertTrue(deserialisedNote.getTags().equals(associatedTags));
     }
 
-    @Test
-    public void toModelType_validCountryNote_returnsCountryNote() {
-        JsonAdaptedNote jsonAdaptedNote = new JsonAdaptedNote(new CountryNote("some note", new Country("SG")));
+    @Test public void toModelType_validCountryNote_returnsCountryNote() throws IllegalValueException {
+        JsonAdaptedNote jsonAdaptedNote =
+                new JsonAdaptedNote(new CountryNote("some note", new Country("SG")));
         assertFalse(jsonAdaptedNote.toModelType().isClientNote());
     }
+    /* todo: future tests:
+    *       1. how to test the @JsonCreator annotated constructor? in the first place it's not used.
+    * */
```

Comment:
formatting issues with this file, and for the testing of the JsonCreator, you can refer to JsonAdaptedClientTest.java, also it only appears to be not used because a third party library (jackson), and not our own code, is calling it, just like how the FXML notation variables don't seemed to be used

Reply:
psps, autoformatter messed up my formatting, I havenow correctly changed it

btw the `JsonAdaptedClientTest.java` doesn't have tests for the JsonCreater right...

![image](https://user-images.githubusercontent.com/38996397/97249883-b98d8d00-183f-11eb-946c-27c7a3779a03.png)